Retrovizor.Web/Controllers/EventController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Retrovizor.Data.Entities.Models;
+using Retrovizor.Domain.Repositories.Interfaces;
+
+namespace Retrovizor.Web.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EventController : ControllerBase
+    {
+        public EventController(IEventRepository eventRepository)
+        {
+            _eventRepository = eventRepository;
+        }
+        private readonly IEventRepository _eventRepository;
+
+        [HttpPost("add")]
+        public IActionResult AddEvent(Event eventToAdd)
+        {
+            var wasAddSuccessful = _eventRepository.AddEvent(eventToAdd);
+
+            if(wasAddSuccessful)
+                return Ok();
+
+            return Forbid();
+        }
+
+        [HttpPost("edit")]
+        public IActionResult EditEvent(Event editedEvent)
+        {
+            var wasEditSucessful = _eventRepository.EditEvent(editedEvent);
+
+            if(wasEditSucessful)
+                return Ok();
+
+            return NotFound();
+        }
+
+        [HttpDelete("delete/{id}")]
+        public IActionResult DeleteEvent(int id)
+        {
+            var wasDeleteSucessful = _eventRepository.DeleteEvent(id);
+
+            if(wasDeleteSucessful)
+                return Ok();
+
+            return NotFound();
+        }
+
+        [HttpGet("get/{id}")]
+        public IActionResult GetEventById(int id)
```

Comment:
GetEventById bi triba vracat samo evente iz odredene autoskole, jer ne zelimo da jedna autoskola pristupi eventima druge, sta je u ovom slucaju moguce. Negdi bi tribala postojat provjera jeli se autoskola podudara s eventom, sta bi se tribalo gledat priko studenta jer event i autoskola nisu direktno povezani. U slucaju da se ne podudara vratit Forbid

Reply:
Trenutno pokusavan skuzit kako cu odradjivat autentifikaciju, kad buden zna u kojen obliku mi dolaze podaci i kako mogu to osigurat prominit cu, al mi to nije problem za sad